### ISSUE TITLE
`prepx analyze` requires `tree` which is not a dependency

### ISSUE THREAD

**Issue by j0ma** [j0ma/prepx#3]
If a user does not have `tree` installed, `prepx analyze` will not work.
This is reproducible using the `nonexistent-tree-executable` branch where the `tree` executable has been [changed](https://github.com/j0ma/prepx/compare/main...nonexistent-tree-executable?diff=unified#diff-4216c66c631c9843073e2c26e83f1deabf8843d87a84f34df6e7be1c3116b5bfL101-R101).

**Error produced:**
```
Traceback (most recent call last):
  File "/home/jonne/miniconda3/envs/prepx-test/bin/prepx", line 33, in <module>
    sys.exit(load_entry_point('prepx', 'console_scripts', 'prepx')())
  File "/home/jonne/miniconda3/envs/prepx-test/lib/python3.9/site-packages/click-8.1.3-py3.9.egg/click/core.py", line 1130, in __call__
    return self.main(*args, **kwargs)
  File "/home/jonne/miniconda3/envs/prepx-test/lib/python3.9/site-packages/click-8.1.3-py3.9.egg/click/core.py", line 1055, in main
    rv = self.invoke(ctx)
  File "/home/jonne/miniconda3/envs/prepx-test/lib/python3.9/site-packages/click-8.1.3-py3.9.egg/click/core.py", line 1657, in invoke
    return _process_result(sub_ctx.command.invoke(sub_ctx))
  File "/home/jonne/miniconda3/envs/prepx-test/lib/python3.9/site-packages/click-8.1.3-py3.9.egg/click/core.py", line 1404, in invoke
    return ctx.invoke(self.callback, **ctx.params)
  File "/home/jonne/miniconda3/envs/prepx-test/lib/python3.9/site-packages/click-8.1.3-py3.9.egg/click/core.py", line 760, in invoke
    return __callback(*args, **kwargs)
  File "/data/coding/prepx/prepx/cli.py", line 102, in analyze_experiment
    pid = sp.run(args_, capture_output=True)
  File "/home/jonne/miniconda3/envs/prepx-test/lib/python3.9/subprocess.py", line 505, in run
    with Popen(*popenargs, **kwargs) as process:
  File "/home/jonne/miniconda3/envs/prepx-test/lib/python3.9/subprocess.py", line 951, in __init__
    self._execute_child(args, executable, preexec_fn, close_fds,
  File "/home/jonne/miniconda3/envs/prepx-test/lib/python3.9/subprocess.py", line 1821, in _execute_child
    raise child_exception_type(errno_num, err_msg, err_filename)
FileNotFoundError: [Errno 2] No such file or directory: 'nonexistent_tree_executable'
```

**Comment by j0ma** [comment]
**Potential fixes**
- Bundle `tree` as a statically linked dependency
- Use `conda` to install it along with other dependencies
    - Could maybe distribute an environment config file
- Just write your own traversal :)
    - This could involve using something like [`rich.tree.Tree`](https://rich.readthedocs.io/en/stable/tree.html)

**Comment by j0ma** [comment]
Custom traversal implemented in in 807a3eea635d785a01b44d8b5591605c70513bdb
